### ISSUE TITLE
About Large Resolution

### ISSUE THREAD

**Issue by LiewFeng** [FiveLu/stream3dppe#3]
Hi, do you conduct experiments with large resolution, such as 1408*512? I always get NaN loss at around 10000 iterations. Should I modify any hyper-parameters for depth supervision?

**Comment by drilistbox** [comment]
We encountered the same experimental phenomenon, which is quite strange. After changing the image resolution in the configuration(https://github.com/FiveLu/stream3dppe/blob/main/projects/configs/StreamPETR_3dppe/streampetr_3dppe_vov_flash_800_bs2_seq_24e_4x4_no_context_womv.py) to 1600x640, we observed NaN values in the loss after 3-4 epochs. Although the loss decreases faster compared to the case of 800x320 input resolution, and training with 1600x640 image resolution on 3DPPE works perfectly fine.

However, due to limited resources, we didn't continue to investigate these issues. Could you please try adjusting the parameters?

**Comment by LiewFeng** [comment]
Sorry, I don't have enough resources, too.

**Comment by jjlinghu** [comment]
This phenomenon disturbs me a lot! When I conduct experiments with large resolution, I also always get unstable training, and I don't know how to deal with it...

**Comment by drilistbox** [comment]
> This phenomenon disturbs me a lot! When I conduct experiments with large resolution, I also always get unstable training, and I don't know how to deal with it...

Given that I am self-funding the rental of GPUs on www.autodl.com for conducting experiments, I believe that if you have access to GPUs, we can collaborate on discussing and debugging together, with the aim of resolving this issue. 

Renting GPUs for each experiment is quite expensive, and I hope you can understand the financial constraints.

**Comment by jjlinghu** [comment]
> > This phenomenon disturbs me a lot! When I conduct experiments with large resolution, I also always get unstable training, and I don't know how to deal with it...
> 
> Given that I am self-funding the rental of GPUs on [www.autodl.com](http://www.autodl.com) for conducting experiments, I believe that if you have access to GPUs, we can collaborate on discussing and debugging together, with the aim of resolving this issue.
> 
> Renting GPUs for each experiment is quite expensive, and I hope you can understand the financial constraints.

Of course, I understand this financial constraints. I am very glad to have the opportunity to communicate with you, an excellent author of this outstanding work!  If you don't mind, could you contact me? Or we can have more detailed communication with other suitable way.  Anyway, this is my WeChat: lhjj_Dec
Sincerly !

**Comment by drilistbox** [comment]
> > > This phenomenon disturbs me a lot! When I conduct experiments with large resolution, I also always get unstable training, and I don't know how to deal with it...
> > 
> > 
> > Given that I am self-funding the rental of GPUs on [www.autodl.com](http://www.autodl.com) for conducting experiments, I believe that if you have access to GPUs, we can collaborate on discussing and debugging together, with the aim of resolving this issue.
> > Renting GPUs for each experiment is quite expensive, and I hope you can understand the financial constraints.
> 
> Of course, I understand this financial constraints. I am very glad to have the opportunity to communicate with you, an excellent author of this outstanding work! If you don't mind, could you contact me? Or we can have more detailed communication with other suitable way. Anyway, this is my WeChat: lhjj_Dec Sincerly !

Hi, I have added your WeChat.

**Comment by Kanon777** [comment]
do you have any progress of this problem? this phenomenon  confused me as well

**Comment by drilistbox** [comment]
> do you have any progress of this problem? this phenomenon confused me as well

Many thanks to the effort of @jjlinghu , we use a cheap way to sovle the problem and will release code in this week.

**Comment by Kanon777** [comment]
> > do you have any progress of this problem? this phenomenon confused me as well
> 
> Many thanks to the effort of @jjlinghu , we use a cheap way to sovle the problem and will release code in this week.

looking forward to it

**Comment by Kanon777** [comment]
> > do you have any progress of this problem? this phenomenon confused me as well
> 
> Many thanks to the effort of @jjlinghu , we use a cheap way to sovle the problem and will release code in this week.

many thanks, looking forward to it

**Comment by Kanon777** [comment]
> > do you have any progress of this problem? this phenomenon confused me as well
> 
> Many thanks to the effort of @jjlinghu , we use a cheap way to sovle the problem and will release code in this week.

hi, when will the code be released?  looking forward to it

**Comment by happygds** [comment]
> > do you have any progress of this problem? this phenomenon confused me as well
> 
> Many thanks to the effort of @jjlinghu , we use a cheap way to sovle the problem and will release code in this week.

Hi, how do you solve this ? I only find fp32 training is stable, but the training speed is too slow.

**Comment by drilistbox** [comment]
For large resolution, we found that the training is stable when csgb is not used, but it get unstable when the traing time is longer as csgb is used.

So we did not solve the problem.  I am sorry.

**Comment by drilistbox** [comment]
> > > do you have any progress of this problem? this phenomenon confused me as well
> > 
> > 
> > Many thanks to the effort of @jjlinghu , we use a cheap way to sovle the problem and will release code in this week.
> 
> Hi, how do you solve this ? I only find fp32 training is stable, but the training speed is too slow.

Did your experimental performance improve?

**Comment by happygds** [comment]
> > > > do you have any progress of this problem? this phenomenon confused me as well
> > > 
> > > 
> > > Many thanks to the effort of @jjlinghu , we use a cheap way to sovle the problem and will release code in this week.
> > 
> > 
> > Hi, how do you solve this ? I only find fp32 training is stable, but the training speed is too slow.
> 
> Did your experimental performance improve?

NAN only occurs in large resolution and I didn't do performance comparison.
